### PR TITLE
Wp hyperdb

### PR DESCRIPTION
**Pull request by bmackinney**
- [x] @rachelwhitton please test (I'll add you to an elite site)
- [x] @nataliejeremy please edit
- [ ] @joshkoenig please :+1: after ^^

### CODE REVIEW

**Comment by rachelwhitton** [comment]
@bmackinney - I revised the draft quite a bit, ready for your review. 

@timani can you update the PR with your test results?

**Comment by rachelwhitton** [comment]
Timani via slack: 
> updated with minimal testing on views but needs more testing.

```
if (isset($_SERVER['PANTHEON_ENVIRONMENT']) && $_SERVER['PANTHEON_ENVIRONMENT'] === 'live') {
  $databases['default']['slave'][] = array(
    'driver' => 'mysql',
    'database' => $_ENV['REPLICA_DB_NAME'],
    'username' => $_ENV['REPLICA_DB_USER'],
    'password' => $_ENV['REPLICA_DB_PASSWORD'],
    'host' => $_ENV['REPLICA_DB_HOST'],
    'port' => $_ENV['REPLICA_DB_PORT'],
  );
}
// Pushing config to Live to verify is time consuming. Making this local for Debug/Development
else {
  $databases['default']['slave'][] = array(
    'driver' => 'mysql',
    'database' => 'pantheon',
    'username' => 'pantheon',
    'password' => 'xxxx',
    'port' => '1700',
    'host' => 'dbserver.live.123-456-789-0f.drush.in',
  );
}
```
Documentation for Drupal is recommended, see: https://www.evernote.com/shard/s389/sh/9edafa85-65d9-4c79-b25e-f5593c7dd182/86a9eaa2edbc6ee1342d496744acd49e

Document SAVEQUERIES for WP to help with logging/debugging?

The Drupal content can be handled in another issue or this PR, I defer to you on that @bmackinney

**Comment by bmackinney** [comment]
@joshkoenig please review/+1?

**Comment by joshkoenig** [comment]
+1 for an equivalent "Using replica Databases with Drupal" doc page. Cross link the two once that's done.

File: source/docs/articles/wordpress/hyperdb.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+title: Scaling WordPress with MySQL Replication
+description: Learn how to distribute requests to replica MySQL databases on WordPress using HyperDB.
+---
+
+Replica MySQL database configuration via the [HyperDB](https://wordpress.org/support/plugin/hyperdb) plugin is available only for WordPress sites on the [Elite service level](/resources/elite-plan-overview). [Contact sales](/pantheon-elite-plans) to inquire about our Elite plans.
```
Comment (joshkoenig) [inline source/docs/articles/wordpress/hyperdb.md]: This should be an alert box. It should read:

"Replica MySQL databases are available for sites on the [Elite service level](/resources/elite-plan-overview). [Contact sales](/pantheon-elite-plans) to inquire about our Elite plans."

File: source/docs/articles/wordpress/hyperdb.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+---
+title: Scaling WordPress with MySQL Replication
```
Comment (joshkoenig) [inline source/docs/articles/wordpress/hyperdb.md]: title: Scaling WordPress with MySQL Replicas and HyperDB

File: source/docs/articles/wordpress/hyperdb.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+### Pass Frontend Read Queries to Replica, WordPress Dashboard Reads and Writes to Master
+Pass all front-end database read queries to the replica, leaving the master dedicated to WordPress dashboard reads and writes. This can better ensure WordPress dashboard availability during high front-end load.
+<script src="http://gist-it.appspot.com/https://github.com/pantheon-systems/pantheon-settings-examples/blob/master/wordpress/master-write-replica-read.dbconfig.php?footer=minimal"></script>
+
+In this example, you’ll want [some form of cache fallback](/docs/articles/wordpress/installing-redis-on-wordpress/) if reads to the replica begin failing.
+
+### Distributed Reads for Localization or Site Networks
+For those scaling WordPress Multisite, one can allocate a collection of master servers amongst sites on the network. Sites A and B could use different MySQL masters, and continue to share the same codebase. This use case is not yet possible on Pantheon. Secondary master database servers would need to be configured elsewhere.
```
Comment (joshkoenig) [inline source/docs/articles/wordpress/hyperdb.md]: Cut this since it's not possible.

**Comment by bmackinney** [comment]
@joshkoenig all fixed up. multidev edition: http://docs-files-panther.pantheon.io/docs/articles/wordpress/hyperdb/

**Comment by joshkoenig** [comment]
Excellent. +1

**Pull request merged**